app/models/feature_completion.rb

```diff
@@ -0,0 +1,8 @@
+class FeatureCompletion < ActiveRecord::Base
+  belongs_to :split
+  belongs_to :app
+
+  validates :split_id, uniqueness: { scope: :app_id }
+
+  attribute :version, :app_version
```

Comment:
Attributes API for the win. Seems like a really nice REST-impedence-matched story for accepting updates to these per-app per-split feature completion records.